### PR TITLE
Port spline interpolation

### PR DESCRIPTION
**Pull request by sgermanserrano**
Port of `spline_interpolation` as it is required by `obstacle_avoidance_planner`

### CODE REVIEW

File: common/math/spline_interpolation/src/spline_interpolation.cpp [hunk 1]
```diff
@@ -163,12 +163,7 @@ std::vector<double> SplineInterpolator::solveLinearSystemExplicit()
     w.push_back(tmp);
   }
 
-  for (size_t i = N - 2; i > 0; --i) ans[i] = ans[i] - ans[i + 1] * w[i];
-
-  return ans;
-}
-
-void LinearSystemSolver::setCoefficients(
+  for (size_t i = N - 2; i > 0; --i) ans[  <buildtool_depend>catkin</buildtool_depend>
```
Comment (esteve) [inline common/math/spline_interpolation/src/spline_interpolation.cpp:166]: Got something here pasted mistakenly.

**Review by esteve — CHANGES_REQUESTED** [review]

File: common/math/spline_interpolation/src/spline_interpolation.cpp (hunk 1 above)
Comment (sgermanserrano) [reply common/math/spline_interpolation/src/spline_interpolation.cpp]: sorted, oddly enough it compiled fine so must've copied just before committing

**Comment by cvasfi** [comment]
@sgermanserrano  I opened https://github.com/tier4/Pilot.Auto/pull/65 few days before but i can close mine if you would prefer?

**Comment by mitsudome-r** [comment]
I will close this for now since this is a duplicate of https://github.com/tier4/Pilot.Auto/pull/65

**Comment by sgermanserrano** [comment]
Makes sense, thanks for merging #65

**Pull request closed without merging**